modules/puppet/templates/puppet-darwin-run-puppet.sh.erb

```diff
@@ -78,8 +78,15 @@ function email_report {
     ERR_SUBJECT=$1
     ERR_MSG=$2
 
+    RELAY_HOST="<%= @smtp_relay_host %>"
     SENDER="root@${FQDN}"
     RECEIVER="${PUPPET_MAIL}"
+
+    if [[ "${RELAY_HOST}" == 'localhost' ]]; then
+        echo "No email server setup. Skipping email."
+        echo $ERR_MSG
+        return
```

Comment:
@gcoxmoz @escapewindow I was thinking about this, I think it's safe to assume we don't have localhost smtp on the macs.
So I exit early if it's localhost, and don't catch the exception in the python otherwise.

Reply:
I can't review because I can't confirm/deny your assumption, but I like the thought process if you're right.

Reply:
It's certainly possible to set up localhost be a relay to a real mail server. Not sure if the hardware mac builders have localhost smtp set up or not, or if we want them to. It would certainly make alerting easier if we had a .forward that sent mail to release+something@m.c.